tiledb/sm/query/query_condition.cc

```diff
@@ -652,23 +652,35 @@ void QueryCondition::apply_ast_node(
         }
       }
     } else {
-      const auto f = result_tile->frag_idx();
+      if (result_tile == nullptr) {
+        throw QueryConditionException(
+            "Failed to apply query condition due to null '" + field_name +
+            "' ResultTile.");
+      }
+
+      const auto& frag_meta = fragment_metadata[result_tile->frag_idx()];
+      // If the tile_tuple is null it was skipped by this fragment in
+      // ReaderBase::skip_field. This is the case for evolved attributes / dims.
+      const auto tile_tuple = result_tile->tile_tuple(field_name);
+      // Special attributes handle null tile_tuples specifically.
+      // We can skip null tile_tuples only if they are not a special attribute.
+      bool skip_tile_tuple = !ArraySchema::is_special_attribute(field_name) &&
```

Comment:
Is `tile_tuple` always null if this *is* a special attribute?

Reply:
Does this mean that there were possibly more bugs lurking if a query condition is applied to one of the special attributes?

Reply:
Thanks for digging in.  TIL!